tests/unittests/GraphOptzTest.cpp

```diff
@@ -273,16 +273,35 @@ TEST_F(GraphOptz, optimizeBatchNormAfterConvFP16) {
 
   EXPECT_EQ(F_->getNodes().size(), 3);
 
+  optimizedF_ = F_->clone(F_->getName().str() + "_optimized");
   ::glow::convertPlaceholdersToConstants(F_, bindings_, {});
-  ::glow::optimize(F_, CompilationMode::Infer);
-  EXPECT_EQ(F_->getNodes().size(), 2);
+  ::glow::convertPlaceholdersToConstants(optimizedF_, bindings_, {});
+  ::glow::optimize(optimizedF_, CompilationMode::Infer);
 
-  ASSERT_EQ(A->getNumUsers(), 1);
-  Node *newCV = A->getUsers().begin()->getUser();
+  EXPECT_EQ(optimizedF_->getNodes().size(), 2);
+
+  ASSERT_EQ(A->getNumUsers(), 2);
+  auto usersIt = A->getUsers().begin();
+  Node *oldCV = usersIt->getUser();
+  EXPECT_TRUE(llvm::isa<ConvolutionNode>(oldCV));
+  ASSERT_EQ(oldCV->getNumUsers(), 1);
+  Node *oldBN = oldCV->getUsers().begin()->getUser();
+  EXPECT_TRUE(llvm::isa<BatchNormalizationNode>(oldBN));
+  ASSERT_EQ(oldBN->getNumUsers(), 1);
+  Node *oldSave = oldBN->getUsers().begin()->getUser();
+  EXPECT_TRUE(llvm::isa<SaveNode>(oldSave));
+
+  usersIt++;
+  Node *newCV = usersIt->getUser();
   EXPECT_TRUE(llvm::isa<ConvolutionNode>(newCV));
   ASSERT_EQ(newCV->getNumUsers(), 1);
-  Node *save = newCV->getUsers().begin()->getUser();
-  EXPECT_TRUE(llvm::isa<SaveNode>(save));
+  Node *newSave = newCV->getUsers().begin()->getUser();
+  EXPECT_TRUE(llvm::isa<SaveNode>(newSave));
```

Comment:
Is this part necessary?  This sort of shows that the BN node is folding into the CV node, but it is very dependent on the internal implementation.

Reply:
Well, that is the whole point of the optimization that is being tested. :)